erdjava/src/test/java/elrond/TransactionTest.java

```diff
@@ -41,12 +41,12 @@ public void shouldSign() throws Exception {
         transaction.setReceiver(Address.fromBech32("erd1cux02zersde0l7hhklzhywcxk4u9n4py5tdxyx7vrvhnza2r4gmq4vw35r"));
         transaction.setGasPrice(1000000000);
         transaction.setGasLimit(70000);
-        transaction.setData("for the book");
+        transaction.setData("for the book with stake");
         transaction.setChainID("1");
         transaction.sign(wallet);
 
-        String expectedSignature = "1702bb7696f992525fb77597956dd74059b5b01e88c813066ad1f6053c6afca97d6eaf7039b2a21cccc7d73b3e5959be4f4c16f862438c7d61a30c91e3d16c01";
-        String expectedJson = "{'nonce':7,'value':'10000000000000000000','receiver':'erd1cux02zersde0l7hhklzhywcxk4u9n4py5tdxyx7vrvhnza2r4gmq4vw35r','sender':'erd1l453hd0gt5gzdp7czpuall8ggt2dcv5zwmfdf3sd3lguxseux2fsmsgldz','gasPrice':1000000000,'gasLimit':70000,'data':'Zm9yIHRoZSBib29r','chainID':'1','version':1,'signature':'1702bb7696f992525fb77597956dd74059b5b01e88c813066ad1f6053c6afca97d6eaf7039b2a21cccc7d73b3e5959be4f4c16f862438c7d61a30c91e3d16c01'}".replace('\'', '"');
+        String expectedSignature = "096c571889352947f285632d79f2b2ee1b81e7acd19ee20510d34002eba0f999b4720f50211b039dd40914284f84c14eb84815bb045c14dbed036f2e87431307";
+        String expectedJson = "{'nonce':7,'value':'10000000000000000000','receiver':'erd1cux02zersde0l7hhklzhywcxk4u9n4py5tdxyx7vrvhnza2r4gmq4vw35r','sender':'erd1l453hd0gt5gzdp7czpuall8ggt2dcv5zwmfdf3sd3lguxseux2fsmsgldz','gasPrice':1000000000,'gasLimit':70000,'data':'Zm9yIHRoZSBib29rIHdpdGggc3Rha2U=','chainID':'1','version':1,'signature':'096c571889352947f285632d79f2b2ee1b81e7acd19ee20510d34002eba0f999b4720f50211b039dd40914284f84c14eb84815bb045c14dbed036f2e87431307'}".replace('\'', '"');
```

Comment:
Hi! Good find and thanks for your contribution. However, this wouldn't have caused issues because when signing the transaction, the `data` field is in byte array format so any specials characters from the `data` field would have been replaced when converting from string to byte array. 
But for extra safety, we can merge your change. 
Before that, would it be possible that you update the `data` field in this test so it would contain special html characters?

Reply:
Hi @bogdan-rosianu,

I see that data field is encoded as base64.
So, in test, i change the value from "for the book" into "for the book with stake" which become "Zm9yIHRoZSBib29rIHdpdGggc3Rha2U=" after encoding process.

The equal sign at the end of the encoded data field is being escaped by gson.

Thank you

Reply:
Ok, good point got it. 
Thank you!